platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DynamicCallProblemReporting.kt

```diff
@@ -66,17 +68,38 @@ class DefaultDynamicCallProblemReporting : DynamicCallProblemReporting {
     }
 
     override fun leaveDynamicCall(entryPoint: Any) {
+        if (currentThreadState.callStack.isEmpty()) {
```

Comment:
🤔 So `enterDynamicCall` is not called but `leaveDynamicCall` is?

Reply:
It shouldn't be called. I added the check here by intertia. Removed.